libsupport/src/ArrowVisitor.cpp

```diff
@@ -134,92 +134,13 @@ class AppendScalarToBuilderVisitor : public katana::ArrowVisitor {
 
   ResultType AcceptFailed(
       const arrow::Scalar& scalar, arrow::ArrayBuilder* builder) {
-    return KATANA_ERROR(
-        katana::ErrorCode::ArrowError, "no matching type {}, {}",
-        scalar.type->name(), builder->type()->name());
-  }
-};
-
-struct ToArrayVisitor : public katana::ArrowVisitor {
-  // Internal data and constructor
-  const std::vector<std::shared_ptr<arrow::Scalar>>& scalars;
-  ToArrayVisitor(const std::vector<std::shared_ptr<arrow::Scalar>>& input)
-      : scalars(input) {}
-
-  using ResultType = katana::Result<std::shared_ptr<arrow::Array>>;
-
-  using AcceptTypes = std::tuple<katana::AcceptAllArrowTypes>;
-
-  template <typename ArrowType, typename BuilderType>
-  arrow::enable_if_null<ArrowType, ResultType> Call(BuilderType* builder) {
-    KATANA_CHECKED(builder->AppendNulls(scalars.size()));
-    return KATANA_CHECKED(builder->Finish());
-  }
-
-  template <typename ArrowType, typename BuilderType>
-  std::enable_if_t<
-      arrow::is_number_type<ArrowType>::value ||
-          arrow::is_boolean_type<ArrowType>::value ||
-          arrow::is_temporal_type<ArrowType>::value,
-      ResultType>
-  Call(BuilderType* builder) {
-    using ScalarType = typename arrow::TypeTraits<ArrowType>::ScalarType;
-
-    KATANA_CHECKED(builder->Reserve(scalars.size()));
-    for (const auto& scalar : scalars) {
-      if (scalar != nullptr && scalar->is_valid) {
-        const ScalarType* typed_scalar = static_cast<ScalarType*>(scalar.get());
-        builder->UnsafeAppend(typed_scalar->value);
-      } else {
-        builder->UnsafeAppendNull();
-      }
+    if (auto res = builder->AppendScalar(scalar); !res.ok()) {
+      return KATANA_ERROR(
+          katana::ErrorCode::ArrowError, "incompatible types: {}, {}",
+          scalar.type->ToString(), builder->type()->ToString());
```

Comment:
scalar.type(), right?

Reply:
Also, this could use KATANA_CHECKED_CONTEXT, right?

Reply:
arrow::Scalar is the exception, `type` is a member instead of a method